src/app/_components/client/AvatarList/AvatarList.tsx

```diff
@@ -0,0 +1,96 @@
+"use client";
+
+import profileImage from "../../../../../public/assets/userProfile.svg";
+import Avatar from "../Avatar/Avatar";
+import { useClickAvatarList } from "./hooks";
+
+import { motion } from "framer-motion";
+import Image from "next/image";
+
+interface AvatarProps {
+  id: string;
+  avatarImage: string | null;
+  name: string;
+}
+
+export interface AvatarListProps {
+  avatars: AvatarProps[];
+}
+
+const AvatarList = ({ avatars }: AvatarListProps) => {
+  const { toggle, handleClickAvatar, handleClickBackground } = useClickAvatarList();
+
+  const variants = {
+    open: { opacity: 1, x: 0 },
+    closed: { opacity: 0, x: "-100%" },
+  };
+
+  if (avatars.length === 1) {
+    const { id, avatarImage } = avatars[0];
+    return (
+      <Avatar
+        userId={id}
+        avatarImage={avatarImage}
+      />
+    );
+  }
+
+  if (avatars.length === 0) {
+    return;
+  }
+
+  return (
+    <>
+      <div className="relative">
+        <div
+          className="flex items-center cursor-pointer"
+          onClick={handleClickAvatar}
+        >
+          {avatars.slice(0, 2).map(({ id, name, avatarImage }, index) => (
+            <div
+              key={id}
+              className={`relative w-[3rem] h-[3rem] rounded-full overflow-hidden ${index === 1 && "-ml-[2.2rem]"} ${index === 1 && "mt-[1rem]"} ${index === 0 && "z-10"}`}
+            >
+              <Image
+                className="object-cover"
+                src={avatarImage || profileImage}
+                alt={name}
+                sizes="100%"
+                priority
+                fill
+              />
+            </div>
+          ))}
+        </div>
+
+        <motion.ul
+          className="absolute left-0 top-0 flex z-10"
+          initial={{ opacity: 0, x: "-100%" }}
+          animate={toggle ? "open" : "closed"}
+          variants={variants}
+          whileTap={{ scale: 0.9 }}
+        >
+          {avatars.map(({ id, avatarImage }) => (
+            <li
+              className="mr-2 rounded-full overflow-hidden"
+              key={id}
+            >
+              <Avatar
+                avatarImage={avatarImage}
+                userId={id}
+              />
+            </li>
+          ))}
+        </motion.ul>
```

Comment:
[ P2 ]

최적화를 위해 `framer-motion`에서 지원하는 `AnimatePrecense` 기능을 활용해 `exit` animation 을 구현해보는건 어떨까요?

현재는 ul태그가 완전히 unmount가 되는게아닌 트릭과같이 숨겨지는 방식이다 보니 필요할때만 mount시켜줄 수 있는 방법이 좋을것 같다고 생각이 들었어요!

```suggestion
        <AnimatePresence>
          {toggle && (
            <motion.ul
              className="absolute left-0 top-0 flex z-10"
              initial={{ opacity: 0, x: "-100%" }}
              animate={{ opacity: 1, x: 0 }}
              exit={{ opacity: 0, x: "-100%" }}
              whileTap={{ scale: 0.9 }}
            >
              {avatars.map(({ id, avatarImage }) => (
                <li
                  className="mr-2 rounded-full overflow-hidden"
                  key={id}
                >
                  <Avatar
                    avatarImage={avatarImage}
                    userId={id}
                  />
                </li>
              ))}
            </motion.ul>
          )}
        </AnimatePresence>
```

Reply:
`AnimatePrecense`에 대해서는 모르고 있었는데 좋은 내용 공유해 주셔서 감사합니다 !! 👏
좋은 제안인 것 같습니다 수정해놓겠습니다 ㅎㅎ 😊